File: app/src/main/java/com/constraint/vagabond/data/entities/RecreationalArea.java
package com.constraint.vagabond.data;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

@Entity
public final class RecreationalArea {

  @PrimaryKey
  @NonNull
  @SerializedName("RecAreaID")
  @Expose
  public final String recAreaID;

  @SerializedName("RecAreaName")
  @Expose
  public final String recAreaName;

  @SerializedName("RecAreaDescription")
  @Expose
  public final String recAreaDescription;

  @SerializedName("RecAreaDirections")
  @Expose
  public final String recAreaDirections;

  @SerializedName("RecAreaPhone")
  @Expose
  public final String recAreaPhone;

  @SerializedName("RecAreaEmail")
  @Expose
  public final String recAreaEmail;

  @SerializedName("MEDIA")
  @Expose
  @TypeConverters(DataConverter.class)
  public final List<RecAreaMedia> recAreaMediaList;

  public RecreationalArea(
      String recAreaID,
      String recAreaName,
      String recAreaDescription,
      String recAreaDirections,
      String recAreaPhone,
      String recAreaEmail,
      List<RecAreaMedia> recAreaMediaList) {
    this.recAreaID = recAreaID;
    this.recAreaName = recAreaName;
    this.recAreaDescription = recAreaDescription;
    this.recAreaDirections = recAreaDirections;
    this.recAreaPhone = recAreaPhone;
    this.recAreaEmail = recAreaEmail;
    this.recAreaMediaList = recAreaMediaList;
  }

  @NonNull
  @Override
  public String toString() {
    String info = "id: " + recAreaID +
            "\n" + "name: " + recAreaName +
            "\n" + "description: " + recAreaDescription +
            "\n" + "directions: " + recAreaDirections +
            "\n" + "phone: " + recAreaPhone +
            "\n" + "email: " + recAreaEmail +
            "\n" + "media_urls" + recAreaMediaList.toString();
    return info;
  }
}

File: app/src/main/java/com/constraint/vagabond/data/entities/RecreationalArea.kt
package com.constraint.vagabond.data.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters

@Entity
data class RecreationalArea(
        @field:PrimaryKey
        @field:SerializedName("RecAreaID")
        @field:Expose
        val recAreaID: String,
        @field:SerializedName("RecAreaName")
        @field:Expose
        val recAreaName: String,
        @field:SerializedName("RecAreaDescription")
        @field:Expose
        val recAreaDescription: String,
        @field:SerializedName("RecAreaDirections")
        @field:Expose
        val recAreaDirections: String,
        @field:SerializedName("RecAreaPhone")
        @field:Expose
        val recAreaPhone: String,
        @field:SerializedName("RecAreaEmail")
        @field:Expose
        val recAreaEmail: String,
        @field:SerializedName("MEDIA")
        @field:Expose
        @field:TypeConverters(DataConverter::class)
        val recAreaMediaList: List<RecAreaMedia>)
